### DISCUSSION TITLE
Schema verification in op

### DISCUSSION THREAD

**Discussion by bernardocortez** [dagster-io/dagster#18663] [Q&A]
Hi all.
How can ensure the schema of an op's input? Is there a way to provide the op's definition (@op, for example) with the desired schema?

**Comment by sryza** [comment]
Dagster types may be what you're looking for, if you want to perform the check every time the op runs: https://docs.dagster.io/concepts/types

**Reply by bernardocortez** [reply]
Yes, we are using Dagster Types. In our cases, DataFrames. I just wanted to be able to further specify the schema of the dataframe! Like the name of the columns and their data types. Is it possible?

**Reply by sryza** [reply]
This pattern might be closer to what you're looking for? https://docs.dagster.io/guides/dagster/dagster_type_factories
